### PR TITLE
Remove merge block processor as block rewards are already disabled in protocol schedule

### PR DESCRIPTION
**Pull request by jframe**
## PR description
Remove the merge block processor as block rewards are already disabled in the protocol schedule by using skipZeroBlockRewards=true and blockreward=0.

Testing that syncing still works
- [x] Local test using Beku
- [x] Bonsai snapsync on Goerli
- [x] Bonsai snapsync on Sepolia
- [ ] Bonsai snapsync on Mainnet

## Fixed Issue(s)

Part of #5354

### CODE REVIEW

**Comment by jframe** [comment]
Testing that syncing still works
- [ ] Bonsai snapsync on Goerli
- [ ] Bonsai snapsync on Sepolia
- [ ] Bonsai snapsync on Mainnet

File: consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeProtocolSchedule.java [hunk 1]
```diff
@@ -93,7 +93,6 @@ private static ProtocolSpecBuilder applyMergeSpecificModifications(
             (gasCalculator, jdCacheConfig) ->
                 MainnetEVMs.paris(
                     gasCalculator, chainId.orElse(BigInteger.ZERO), EvmConfiguration.DEFAULT))
-        .blockProcessorBuilder(MergeBlockProcessor::new)
```
Comment (gfukushima) [inline consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeProtocolSchedule.java:96]: Isn't the MergeBlockProcessor Optimized in terms of all the rewardCoinbase code path? If we remove this and start calling the rewardCoinbase method from the MainnetBlockProcessor aren't we executing more checks and code than we're supposed to?

Comment (jframe) [reply consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeProtocolSchedule.java:96]: I don't think the MergeBlockProcessor is that optimised. We exit the rewardCoinbase method early due to https://github.com/hyperledger/besu/blob/main/ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetBlockProcessor.java#L55 with two very simple checks as opposed to the single isPostMerge check from the MergeBlockProcessor.

Comment (gfukushima) [reply consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeProtocolSchedule.java:96]: If what you mentioned in the PR description is true: skipZeroBlockRewards=false then that if is probably skipped

File: consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeProtocolSchedule.java [hunk 2]
```diff
@@ -104,11 +103,8 @@ private static ProtocolSpecBuilder applyMergeSpecificModifications(
 
   private static ProtocolSpecBuilder unapplyMergeModificationsFromShanghaiOnwards(
       final ProtocolSpecBuilder specBuilder) {
-    // TODO Withdrawals Get rid of MergeBlockProcessor
     // inherits merge config from MainnetProtocolSpecs.parisDefinition
-    // This would be Function.identify() but MergeBlockProcessor can't be used in
-    // MainnetProtocolSpecs due to circular dependency
-    return specBuilder.blockProcessorBuilder(MergeBlockProcessor::new);
+    return specBuilder;
```
Comment (siladu) [inline consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeProtocolSchedule.java:107]: Wondering if we should use Function::identity instead now? 
I think it's better not to leave this method around to discourage future changes.

I like that this method has a name though...maybe creating a named FunctionalInterface version of identity() would be a good compromise?

**Review by siladu — APPROVED** [review]
Just a nitpick really

File: consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeProtocolSchedule.java (hunk 2 above)
Comment (jframe) [reply consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeProtocolSchedule.java:107]: I left the function just for the name. A comment would probably just as well though. So will remove it add a comment instead.

Comment (jframe) [reply consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeProtocolSchedule.java]: done

File: consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeProtocolSchedule.java (hunk 1 above)
Comment (jframe) [reply consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeProtocolSchedule.java:96]: The skipZeroBlockRewards is set to true from spurious dragon onwards https://github.com/hyperledger/besu/blob/main/ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetProtocolSpecs.java#L267. So for Paris onwards where blockrewards=0 then we return immediately without trying apply rewards.

Comment (gfukushima) [reply consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeProtocolSchedule.java:96]: You're are right, spurious dragon changes that to true, I was misled by the PR description.

**Review by gfukushima — APPROVED** [review]

**Pull request merged**